exercises/twelve-days/test/Tests.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE RecordWildCards #-}
+
+import Data.Foldable     (for_)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+
+import TwelveDays (recite)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "responseFor" $ for_ cases test
+  where
+      test Case{..} = it description assertion
+        where
+          assertion = recite start stop `shouldBe` expected
+
+data Case = Case { description :: String
+                 , start       :: Int
+                 , stop        :: Int
+                 , expected    :: [String]
+                 }
+
+cases :: [Case]
+cases = [ Case { description = "first day a partridge in a pear tree"
+               , start = 1
+               , stop = 1
+               , expected = [ 
+                  "On the first day of Christmas my true love gave to me, a Partridge in a Pear Tree."
+                 ]
+               },
```

Comment:
Personally, I prefer the elm method which is to place the commas before. This was more to just be consistent with what I saw in other haskell tests. I can update it or leave it, but I'll leave it unless explicitly requested to make the alteration.

Reply:
> > In all lists, all list elements got trailing commas. In all record fields, all fields got leading commas. 
>
> This was more to just be consistent with what I saw in other haskell tests.

The only other Haskell test where I see a trailing comma are lens-person and pascals-triangle, BUT I may have missed some because I only grepped the specific strings `},`  and `],` and not any other variants.

Thus:

For consistency with all other tests in this repo, let us have leading commas for the separation between different `Case`.

I think it will also be better to have leading commas separating the strings in `expected`, for the purpose of having this file be consistent with itself. Even if this isn't done, there is still a sort of consistency, though it is odd, so I could be convinced that it should stay. Your option on this second one.